File: DSA_python/DSA01001 XÂU NHỊ PHÂN KẾ TIẾP.py
def update(A):
    for i in range(len(A)-1, -1, -1):
        if A[i] == "0":
            A[i] = "1"
            break
        A[i] = "0"
    return A

def submit():
    n = list(input())
    n = update(n)
    for i in range(len(n)):
        print(n[i], end = "")
    print()

test_num = int(input())
for i in range(test_num):
    submit()

File: DSA/DSA01001 XÂU NHỊ PHÂN KẾ TIẾP.cpp
#include <bits/stdc++.h>
using namespace std;

void update(string &s, int n)
{
    for (int i = n-1; i >= 0; i--)
    {
        if (s[i] == '0')
        {
            s[i] = '1';
            break;
        }
        s[i] = '0';
    }
}

void submit()
{
    string s;
    cin >> s;

    update(s, s.size());
    cout << s << endl;
}   

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();

    while (so_bo_test--)
    {
        submit();
    }

    return 0;
}